### PR TITLE
[SPARK-35909][DOCS] Fix broken Python Links in docs/sql-getting-started.md

### PR DESCRIPTION
**Pull request by dhruvildave**
### What changes were proposed in this pull request?

The hyperlinks in Python code blocks in [Spark SQL Guide - Getting Started](https://spark.apache.org/docs/latest/sql-getting-started.html) currently point to invalid addresses and return 404. This pull request fixes that issue by pointing them to correct links in Python API docs.

### Why are the changes needed?

Error in documentation classifies as a bug and hence needs to be fixed.

### Does this PR introduce _any_ user-facing change?

Yes. This PR fixes documentation error in https://spark.apache.org/docs/latest/sql-getting-started.html

### How was this patch tested?

This patch was locally built after cloning the repo from scratch and then doing a clean build after fixing the required problems.

### CODE REVIEW

File: docs/sql-getting-started.md [hunk 1]
```diff
@@ -41,7 +41,7 @@ The entry point into all functionality in Spark is the [`SparkSession`](api/java
 
 <div data-lang="python"  markdown="1">
 
-The entry point into all functionality in Spark is the [`SparkSession`](api/python/pyspark.sql.html#pyspark.sql.SparkSession) class. To create a basic `SparkSession`, just use `SparkSession.builder`:
+The entry point into all functionality in Spark is the [`SparkSession`](api/python/reference/api/pyspark.sql.SparkSession.html) class. To create a basic `SparkSession`, just use `SparkSession.builder`:
```
Comment (dongjoon-hyun) [inline docs/sql-getting-started.md:44]: Hi, @dhruvildave . Is this correct? In Apache Spark website, the original one seems to work correctly.
- https://spark.apache.org/docs/latest/sql-getting-started.html

Comment (dhruvildave) [reply docs/sql-getting-started.md:44]: Hello! Thanks for the review. I have created a small video showing exactly where the errors are encountered.

https://youtu.be/wIZ56za9zmA

Comment (dongjoon-hyun) [reply docs/sql-getting-started.md:44]: Oh, thanks!

Comment (dongjoon-hyun) [reply docs/sql-getting-started.md:44]: Ah, got it. It's inside Python tab. I missed the point.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @dhruvildave .
Merged to master/3.1/3.0.

**Comment by dongjoon-hyun** [comment]
I added you to the Apache Spark contributor group and assigned SPARK-35909 to you. Welcome! :)

**Comment by HyukjinKwon** [comment]
Thanks @dhruvildave!

**Pull request closed without merging**